tooltester/src/main/java/io/dockstore/tooltester/client/cli/Client.java

```diff
@@ -322,12 +337,56 @@ private void setUpWorkflowApi() {
         defaultApiClient.setBasePath(this.workflowRunnerConfig.getServerUrl());
         setWorkflowsApi(new WorkflowsApi(defaultApiClient));
     }
-    private void runToolTesterOnWorkflows(String pathToConfigFile) throws InterruptedException {
+
+    private List<File> getAllFilesInDirectory(File directory) {
+        List<File> filesInDirectory = new ArrayList<>();
+        for (File file: directory.listFiles()) {
+            if (file.isDirectory()) {
+                filesInDirectory.addAll(getAllFilesInDirectory(file));
+            } else {
+                filesInDirectory.add(file);
+            }
+        }
+        return filesInDirectory;
+    }
+
+    private void uploadResults(String urlToUploadTo, String resultsDirectoryString, String pathToConfigFile) {
+        this.workflowRunnerConfig = new WorkflowRunnerConfig(pathToConfigFile);
+        setUpExtendedGa4GhApi(urlToUploadTo);
+        Gson gson = new Gson();
+
+        Path resultsDirectoryPath = Path.of(resultsDirectoryString);
+        File resultsDirectory = new File(resultsDirectoryString);
+        List <File> allResultFiles = getAllFilesInDirectory(resultsDirectory);
+        for (File file: allResultFiles) {
+
+            Path path = Path.of(file.getPath());
+            String fileContent = null;
+            try {
+                fileContent = Files.readString(path);
+            } catch (IOException e) {
+                exceptionMessage(e, "There was an error reading " + path + " to a string", IO_ERROR);
```

Comment:
Similar comment about whether or not you want to exit the program here when there's more files that can be uploaded

Reply:
It now logs an error message if there is an error and continues on.